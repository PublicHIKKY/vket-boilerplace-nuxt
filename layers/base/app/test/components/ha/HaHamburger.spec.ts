import { mount } from '@vue/test-utils'
import HaHamburger from '#base/app/components/ha/HaHamburger.vue'

test('ref component', () => {
  expect(HaHamburger).toBeTruthy()
})

test('mount component', () => {
  const wrapper = mount(HaHamburger, {
    props: {
      isOpen: false,
    },
  })
  expect(wrapper.getCurrentComponent()).toBeTruthy()
  expect(wrapper.html()).toMatchSnapshot()
})

describe('props', () => {
  it(':isOpen:true', () => {
    const wrapper = mount(HaHamburger, {
      props: {
        isOpen: true,
      },
    })
    expect(wrapper.attributes('class')).toBe('ha-humberger-button -open')
  })
  it(':isOpen:false', () => {
    const wrapper = mount(HaHamburger, {
      props: {
        isOpen: false,
      },
    })
    expect(wrapper.attributes('class')).toBe('ha-humberger-button')
  })
})

describe('emit', () => {
  it(':update:modelValue', async () => {
    const wrapper = mount(HaHamburger, {
      props: {
        isOpen: false,
      },
    })
    await wrapper.trigger('click')
    expect(wrapper.emitted()).toHaveProperty('click')
    expect(wrapper.emitted()['click']).toHaveLength(1)
    // NOTE: voidなので[[]]
    expect(wrapper.emitted()['click']).toEqual([[]])
  })
})
