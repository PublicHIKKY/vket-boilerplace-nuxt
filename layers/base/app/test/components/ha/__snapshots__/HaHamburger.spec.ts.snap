// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `
"<div data-v-398903c3="" class="ha-humberger-button">
  <div data-v-398903c3="" class="line"></div>
</div>"
`;
